lib/shopping_cart.rb

```diff
@@ -0,0 +1,69 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity[0..1].to_i
+    @products = []
+  end
+
+  # puts a product inside of the cart
+  # product -> array of products
+  def add_product(product)
+    @products << product
+  end
+
+  # tells you what the cart's name is and what the cart's total capacity is
+  # no argument -> hash
+  def details
+    {name: name, capacity: capacity }
+  end
+
+  # tells you the total # of products within a cart
+  # no argument -> integer
+  def total_number_of_products
+    products.reduce(0) { |memo, product| memo += product.quantity }
```

Comment:
You could also use `sum` here

Reply:
oh good to know! that would've saved me a ton of typing.